prov/gni/src/gnix_av.c

```diff
@@ -763,7 +764,7 @@ DIRECT_FN int gnix_av_open(struct fid_domain *domain, struct fi_av_attr *attr,
 			goto cleanup;
 		}
 
-		type = attr->type;
+		type = attr->type != FI_AV_UNSPEC ? attr->type : FI_AV_TABLE;
```

Comment:
we already default type to FI_AV_TABLE.
I think type should only be set if attr->type != FI_AV_UNSPEC
otherwise we have two places where we are setting the default